File: src/frontend/tests/fixtures/apphost-languages/go/apphost.go
package main

import (
	"log"

	"apphost/modules/aspire"
)

func main() {
	builder, err := aspire.CreateBuilder()
	if err != nil {
		log.Fatal(aspire.FormatError(err))
	}

	web := builder.
		AddExecutable("web", "node", ".", []string{"service.mjs"}).
		WithHttpEndpoint(&aspire.WithHttpEndpointOptions{Env: aspire.StringPtr("PORT")}).
		WithExternalHttpEndpoints()
	if err := web.Err(); err != nil {
		log.Fatal(aspire.FormatError(err))
	}

	app, err := builder.Build()
	if err != nil {
		log.Fatal(aspire.FormatError(err))
	}
	if err := app.Run(); err != nil {
		log.Fatal(aspire.FormatError(err))
	}
}

File: src/frontend/tests/fixtures/apphost-languages/rust/apphost.rs
#[path = ".aspire/modules/mod.rs"]
mod aspire;

use aspire::*;
use serde_json::Value;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let builder = create_builder(None)?;

    builder.add_external_service(
        "web",
        Value::String("https://aspire.dev".to_string()),
    )?;

    let app = builder.build()?;
    app.run(None)?;
    Ok(())
}
